src/main/range-util.js

```diff
@@ -3,6 +3,24 @@
 const assert = require("assert");
 const comments = require("./comments");
 
+const isJsonParser = ({ parser }) =>
+  parser === "json" || parser === "json5" || parser === "json-stringify";
+
+function findCommonAncestor(startNodeAndParents, endNodeAndParents) {
+  const startNodeAndAncestors = [
+    startNodeAndParents.node,
+    ...startNodeAndParents.parentNodes,
+  ];
+  const endNodeAndAncestors = [
+    endNodeAndParents.node,
+    ...endNodeAndParents.parentNodes,
+  ];
+  return startNodeAndAncestors.find(
+    (node) =>
+      jsonSourceElements.has(node.type) && endNodeAndAncestors.includes(node)
```

Comment:
Nested loops. It's better to make `endNodeAndAncestors` a set instead of an array.